cargo-dist/src/init.rs

```diff
@@ -84,15 +128,59 @@ pub fn do_init(cfg: &Config, args: &InitArgs) -> DistResult<()> {
         }
     };
 
+    // If we're migrating, the configuration will be missing the
+    // generic workspace specification, and will have some
+    // extraneous cargo-specific stuff that we don't want.
+    if is_migrating {
+        // Remove Cargo-specific workspace
+        workspace_toml.remove("workspace");
+
+        // Write generic workspace config
+        let mut table = toml_edit::table();
+        if let Some(t) = table.as_table_mut() {
+            let mut array = toml_edit::Array::new();
+            array.push("cargo:.");
+            t["members"] = toml_edit::value(array);
+            t.set_position(0);
+        }
+        workspace_toml.insert("workspace", table);
+
+        // These fields will continue to be read from Cargo.toml
+        workspace_toml.remove("dependencies");
+        workspace_toml.remove("dev-dependencies");
+        workspace_toml.remove("package");
+        workspace_toml.remove("profile");
```

Comment:
We definitely can't play whackamole here. We know what we want is under `workspace.metadata.dist` and/or `package.metadata.dist`, why can't we just precisely grab that?

Reply:
Oh yes, good point. Not sure why I didn't think in terms of "grab what we want" instead of "delete what we don't want".